docs/predictive-analytics/model-retraining/process.sql

```diff
@@ -0,0 +1,66 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Create stream of predictions
+CREATE STREAM PREDICTED_WEIGHT(
+  "Fish_Id" VARCHAR KEY,
+  "Species" VARCHAR,
+  "Height" DOUBLE,
+  "Length" DOUBLE,
+  "Prediction" DOUBLE
+  )
+WITH(
+  KAFKA_TOPIC = 'kt.mdb.weight-prediction', 
+  VALUE_FORMAT = 'JSON',
+  PARTITIONS = 6
+);
+
+-- Create stream of actual weights
+CREATE STREAM ACTUAL_WEIGHT(
+  "Fish_Id" VARCHAR KEY,
+  "Species" VARCHAR,
+  "Weight" DOUBLE
+  )
+WITH(
+  KAFKA_TOPIC = 'kt.mdb.machine-weight', 
+  VALUE_FORMAT = 'JSON',
+  PARTITIONS = 6
+);
+
+-- Create stream joining predictions with actual weights
+CREATE STREAM DIFF_WEIGHT
+WITH(
+  KAFKA_TOPIC = 'diff_weight', 
+  VALUE_FORMAT = 'JSON'
+)
+AS SELECT
+   'Key' AS "Key",
+   PREDICTED_WEIGHT."Fish_Id" AS "Fish_Id",
+   PREDICTED_WEIGHT."Species" AS "Species",
+   PREDICTED_WEIGHT."Length" AS "Length",
+   PREDICTED_WEIGHT."Height" AS "Height",
+   PREDICTED_WEIGHT."Prediction" AS "PredictedWeight",
+   ACTUAL_WEIGHT."Weight" AS "ActualWeight",
+   ROUND(ABS(PREDICTED_WEIGHT."Prediction" - ACTUAL_WEIGHT."Weight") / ACTUAL_WEIGHT."Weight", 3) AS "Error"
+FROM PREDICTED_WEIGHT
+INNER JOIN ACTUAL_WEIGHT
+WITHIN 1 MINUTE
```

Comment:
We should set a `GRACE` period (we plan to make `GRACE` mandatory so it's best to start using is right away -- this is something we might want to double check in all recipes?)

Reply:
Done, at least for this recipe.